### PR TITLE
Add dashboard widget

### PR DESCRIPTION
**Pull request by seothemes**
## Description

Adds a new widget to the WordPress dashboard screen `/wp-admin/index.php`.

Please note that these changes require assets to be recompiled with `npm run build:scripts:legacy && npm run build:styles`

## How has this been tested?

Manually

## Screenshots

first pass dashboard widget design

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code has been tested.
- [x] My code passes all existing automated tests.
- [x] My code follows the LifterLMS Coding & Documentation Standards.

### CODE REVIEW

File: assets/js/llms-analytics.js [hunk 1]
```diff
@@ -7,7 +7,7 @@
  * @since 3.36.3 Added the `allow_clear` paramater when initializiing the `llmsStudentSelect2`.
  * @since 4.3.3 Legends will automatically display on top of the chart.
  * @since 4.5.1 Show sales reporting currency symbol based on LifterLMS site options.
- *
+ * @version [version]
```
Comment (eri-trabiccolo) [inline assets/js/llms-analytics.js:10]:
```suggestion
 * @version [version]
 *
```

File: assets/js/llms-analytics.js [hunk 2]
```diff
@@ -203,13 +203,14 @@
 		 * @param    obj   $widget  jQuery selector of the widget element
 		 * @return   void
 		 * @since    3.0.0
-		 * @version  3.16.8
+		 * @since [version] Change h1 tag to .llms-widget-content.
+		 * @version [version]
```
Comment (eri-trabiccolo) [inline assets/js/llms-analytics.js]:
```suggestion
```
Also if you wanna take the occasion to improve the docblock of this method... :D <3

File: includes/admin/class.llms.admin.dashboard-widget.php [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+/**
+ * Admin Dashboard Widget class.
+ *
+ * @since [version]
+ */
+class LLMS_Admin_Dashboard_Widget {
+
+	/**
+	 * Constructor
```
Comment (eri-trabiccolo) [inline includes/admin/class.llms.admin.dashboard-widget.php]:
```suggestion
	 * Constructor.
```

File: includes/admin/class.llms.admin.dashboard-widget.php [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function __construct() {
+		add_action( 'wp_dashboard_setup', array( $this, 'add_dashboard_widget' ) );
+	}
+
+	/**
+	 * Add the dashboard widget
```
Comment (eri-trabiccolo) [inline includes/admin/class.llms.admin.dashboard-widget.php]:
```suggestion
	 * Add the dashboard widget.
```

File: includes/admin/class.llms.admin.dashboard-widget.php [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+	public function add_dashboard_widget() {
+		wp_add_dashboard_widget(
+			'llms_dashboard_widget',
+			'LifterLMS ' . __( 'Quick Links', 'lifterlms' ),
+			array( $this, 'output' )
+		);
+	}
+
+	/**
+	 * Output the dashboard widget
```
Comment (eri-trabiccolo) [inline includes/admin/class.llms.admin.dashboard-widget.php]:
```suggestion
	 * Output the dashboard widget.
```

File: templates/admin/reporting/tabs/widgets.php [hunk 6]
```diff
@@ -27,7 +28,14 @@
...
 
-				<h1><?php echo $opts['content']; ?></h1>
+				<?php
+				printf(
+					'<%s class="llms-widget-content">%s</%s>',
+					esc_html( $opts['content_tag'] ?? 'h3' ),
+					esc_html( $opts['content'] ?? '' ),
+					esc_html( $opts['content_tag'] ?? 'h3' )
+				);
+				?>
```
Comment (eri-trabiccolo) [inline templates/admin/reporting/tabs/widgets.php:40]: should update the version of this file

File: includes/admin/class.llms.admin.assets.php [hunk 7]
```diff
@@ -458,15 +463,15 @@ protected function get_analytics_options() {
 	 */
 	protected function maybe_enqueue_reporting( $screen ) {
 
-		if ( in_array( $screen->base, array( 'lifterlms_page_llms-reporting', 'lifterlms_page_llms-dashboard' ), true ) ) {
+		if ( in_array( $screen->base, array( 'lifterlms_page_llms-reporting', 'lifterlms_page_llms-dashboard', 'dashboard' ), true ) ) {
```
Comment (eri-trabiccolo) [inline includes/admin/class.llms.admin.assets.php:470]: missing docblock's changelog update

**Review by eri-trabiccolo — CHANGES_REQUESTED** [review]
doc changes.
Also this doesn't seem to correctly work: the widgets in the dashboard page do not show up, loading spinner forever.

Looks like a js error: datepicker missing, we need to enqueue jquery-datepicker on that page, or change the analytics js so to not break when the datepicker is not there. We actually don't need a datepicker tho only show those widgets...
`Uncaught TypeError: $(...).datepicker is not a function [llms-analytics.js:66:28]`

File: assets/js/llms-analytics.js [hunk 8]
```diff
@@ -63,10 +64,12 @@
 		 */
 		this.bind = function() {
 
-			$( '.llms-datepicker' ).datepicker( {
-				dateFormat: 'yy-mm-dd',
-				maxDate: 0,
-			} );
+			if ( $( '.llms-datepicker' ).length && $.fn.datepicker ) {
```
Comment (eri-trabiccolo) [inline assets/js/llms-analytics.js:68]: I would add a changelog to the `bind()` docblock to log the change.

File: assets/js/llms-analytics.js [hunk 9]
```diff
@@ -200,11 +203,12 @@
...
 		 * Load a specific widget
 		 *
-		 * @param    obj   $widget  jQuery selector of the widget element
-		 * @return   void
-		 * @since    3.0.0
+		 * @since 3.0.0
 		 * @since [version] Change h1 tag to .llms-widget-content.
 		 * @version [version]
+		 *
+		 * @param obj $widget jQuery selector of the widget element
```
Comment (eri-trabiccolo) [inline assets/js/llms-analytics.js]:
```suggestion
		 * @param obj $widget The jQuery selector of the widget element.
```

File: assets/js/llms-analytics.js (hunk 2 above)
Comment (eri-trabiccolo) [reply assets/js/llms-analytics.js]: I'd remove the version here, we use the `version` tag on file headers only.

**Review by eri-trabiccolo — CHANGES_REQUESTED** [review]
docs <3

**Review by eri-trabiccolo — APPROVED** [review]

**Review by kimcoleman — APPROVED** [review]

**Comment by eri-trabiccolo** [comment]
Skipping this for 7.1.3, because it's a minor change, will be merged in 7.2

**Pull request merged**